packages/server/lib/controllers/proxy.controller.ts

```diff
@@ -294,13 +294,31 @@ See https://docs.nango.dev/guides/proxy#proxy-requests for more information.`
      * @param {AxiosError} error
      * @param {attemptNumber} number
      */
-    private retry = async (activityLogId: number, error: AxiosError, attemptNumber: number): Promise<boolean> => {
+    private retry = async (activityLogId: number, config: ProxyBodyConfiguration, error: AxiosError, attemptNumber: number): Promise<boolean> => {
+        if (config.template.proxy && config.template.proxy.retry_header) {
+            const retryHeader = config.template.proxy.retry_header;
+            const retryHeaderVal = error?.response?.headers[retryHeader] || error?.response?.headers[retryHeader.toLowerCase()];
```

Comment:
Good feedback!

Since this is on a per provider basis we'll tackle different measurement units as it comes up per providers – for now we'll treat this as the v1 of handling rate limits via headers. I updated the interface so you can specify `at` or `after` under the `proxy.retry` propery:

````
discord:
    auth_mode: OAUTH2
    authorization_url: https://discord.com/api/oauth2/authorize
    token_url: https://discord.com/api/oauth2/token
    authorization_params:
        response_type: code
    proxy:
        retry:
            after: 'Retry-After'
```

```
github:
    auth_mode: OAUTH2
    authorization_url: https://github.com/login/oauth/authorize
    token_url: https://github.com/login/oauth/access_token
    proxy:
        base_url: https://api.github.com
        retry:
            at: 'x-ratelimit-reset'
    docs: https://docs.github.com/en/rest
```

Let me know if this covers your usage cases and any other feedback! Thanks again for the quick response!

Reply:
Cool, let's keep it like this in v1, that should cover our use-case.